fluent/migrate/repo_client.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+from typing import Tuple
+
+import json
+from subprocess import run
+
+from os.path import isdir, join
+
+import hglib
+
+
+def git(root: str, *args: str) -> str:
+    """
+    Wrapper for calling command-line git in the `root` directory.
+    Raises an exception on any error, including a non-0 return code.
+    Returns the command's stdout as a string.
+    """
+    git = ["git"]
+    git.extend(args)
+    proc = run(git, capture_output=True, cwd=root, encoding="utf-8")
+    if proc.returncode != 0:
+        raise Exception(proc.stderr or f"git {args[0]} failed")
```

Comment:
This would log only the first parameter (e.g. `add`, `commit`)? If so, should we log the entire command instead?

Reply:
I would be surprised if the `or` condition here ever failed for the first part, so the issue is pretty academic. When failing, `git` is really rather good about telling you how/why.

But sure, we could to this:
```suggestion
        raise Exception(proc.stderr or f"git command failed: {args}")
```

Please accept this suggestion if you'd prefer that.